linkml/generators/fhirgen.py

```diff
@@ -0,0 +1,393 @@
+from linkml.generators.docgen import _ensure_ranked
+from linkml.utils.generator import Generator
+
+
+# todo: make some of these class properties?
+FHIRGEN_DIR = os.path.join(os.path.dirname(__file__), 'fhirgen')
+MODEL_MAPPING_PATH = os.path.join(FHIRGEN_DIR, 'mapping__enum_valueset.csv')
+FHIR_VALUESET_FIELD_MUTATIONS: Dict[str, Callable] = {
+    'status': lambda x: 'unknown'  # todo: temp
+}
+
+
+class FhirValueSetGenerator(Generator):
+    """Generate FHIR ValueSet JSON from a LinkML YAML Enum schema"""
+    generatorname = os.path.basename(__file__)
+    generatorversion = "0.0.1"
+    visit_all_class_slots = False
+    genmeta = False  # todo: Do I need this?
+
+    def __init__(
+        self,
+        schema: Union[str, TextIO, SchemaDefinition],
+        outpath: str = None,
+        genmeta: bool = False,
+    ) -> None:
+        """
+        Creates a generator object that can write output
+
+        :param schema: path to schema file or schema object
+        :param outpath: path to save file or file(s)
+        :param genmeta:
+        """
+        self.sourcefile = schema
+        self.schemaview = SchemaView(schema)
+        self.schema = self.schemaview.schema
+        self.format = 'json'  # todo: support w/ CLI
+        self.outpath = outpath
+        self.genmeta = genmeta
+
+    def serialize(self, outpath: str = None) -> None:
+        """
+        Serialize Enums in schema to JSON
+
+        :param directory: relative or absolute path to directory in which documents are to be written
+        :param outpath: path to save file or file(s)
+        :return:
+        """
+        sv = self.schemaview
+        enums: Dict[str, EnumDefinition] = sv.all_enums()
+        # TODO: This is super temporary; has to do with unit test and not all enums being well formed:
+        if 'EmploymentEventType' in enums:
+            enums = {'EmploymentEventType': enums['EmploymentEventType']}
+        # todo: `en` and `n` below are redundant?
+        for en, e in enums.items():
+            n = self.name(e)
+            d: Dict = self.serialize_enum(n, e)
+            out_str: str = json.dumps(d)
+            if outpath:
+                self._write(out_str, outpath, n)
+
+    @staticmethod
+    def model_mappings(path: str = MODEL_MAPPING_PATH) -> Dict[str, str]:
+        """Get model mappings from CSV"""
+        with open(path) as csvfile:
+            reader = csv.reader(csvfile, delimiter=',')
+            rows = [x for x in reader][1:]  # don't need header
+        d = {
+            row[0]: row[1]  # easiest way; assumes linkml is always col 1, and fhir is col 2
+            for row in rows
+        }
+        d = {k: v for k, v in d.items() if k}  # removes anything left over from empty CSV rows
+
+        return d
+
+    # TODO: Most of the work should go here
+    # todo: Since I want to return a dict, should I call this dictify instead of serialize?
+    def serialize_enum(self, enum_name: str, enum: EnumDefinition) -> Dict:
```

Comment:
FYI: This is where most of the work will go, apart from the mappings within the CSV.